flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/channel/ChannelStateWriteRequest.java

```diff
@@ -52,6 +53,10 @@ static ChannelStateWriteRequest write(long checkpointId, InputChannelInfo info,
 		return buildWriteRequest(checkpointId, "writeInput", iterator, (writer, buffer) -> writer.writeInput(info, buffer));
 	}
 
+	static ChannelStateWriteRequest write(long checkpointId, ResultSubpartitionInfo info, Buffer... buffers) {
+		return buildWriteRequest(checkpointId, "writeOutput", ofElements(Buffer::recycleBuffer, buffers), (writer, buffer) -> writer.writeOutput(info, buffer));
+	}
+
```

Comment:
What's the difference to the previous version? Especially in the behaviour? It looks like a subtle change that I'm missing.

Reply:
It is using `CloseableIterator` now, which recycles on unused elements on `close()`. 
Previous version recycled all buffers in `cancel` regardless of whether they were used or not.